domain/consensus/processes/blockprocessor/validateandinsertblock.go

```diff
@@ -116,7 +117,8 @@ func (bp *blockProcessor) validateAndInsertBlock(block *externalapi.DomainBlock,
 		return nil, err
 	}
 
-	log.Debugf("Block %s validated and inserted", blockHash)
+	hashrate := util.GetHashrateString(util.CompactToBig(block.Header.Bits()), bp.targetTimePerBlock)
```

Comment:
This is a bit of a heavy operation to do for every block. Please add it to the log closure below so that it doesn't run at higher log levels.

Reply:
I was trying to think how heavy it is, and I thought that if we think it is heavy then we can move it next to the PoW validation where we already have the target in `big.Int` but then it will be its own log.

I can convert this log also to a log closure, but the log closure below is about the virtual, not the block we just received (and I want to see this information also in headers sync)

Reply:
> I can convert this log also to a log closure
Sounds good, please do.